sympy/simplify/tests/test_cse.py

```diff
@@ -357,3 +357,9 @@ def test_issue_7840():
     assert new_eqn[0] == expr
     # there should not be any replacements
     assert len(substitutions) < 1
+
+def test_issue_8891():
+    m1 = Matrix([w + x, w + x])
+    m2 = Matrix([y + z, y + z])
+    ans = '([(x0, w + x), (x1, y + z)], [[Matrix([\n[x0],\n[x0]])], [Matrix([\n[x1],\n[x1]])]])'
+    assert str(cse([m1, m2])) == ans
```

Comment:
Doing this str() test is ok but not ideal because it misses checking anything else about the types. You should be testing to see if the objects are equivalent.